includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-usage-reports.php

```diff
@@ -25,63 +25,38 @@ private static function get_mc_instance() {
 	/**
 	 * Creates a usage report.
 	 *
-	 * @param int $last_n_days Number of days to get the report for.
-	 *
 	 * @return array Usage report.
 	 */
-	public static function get_usage_report( $last_n_days ) {
+	public static function get_usage_report() {
 		$mailchimp_instance = self::get_mc_instance();
 		$mc_api             = new Mailchimp( $mailchimp_instance->api_key() );
 
-		// Get daily activity for each list.
-		$lists = $mc_api->get( 'lists', [ 'count' => 1000 ] );
+		$report = new Newspack_Newsletters_Service_Provider_Usage_Report();
+		$lists  = $mc_api->get( 'lists', [ 'count' => 1000 ] );
+
 		foreach ( $lists['lists'] as &$list ) {
+			$report->total_contacts += $list['stats']['member_count'];
+			
+			// Get daily activity for each list.
 			$activity_response = $mc_api->get(
 				'lists/' . $list['id'] . '/activity',
 				[
-					'count' => $last_n_days,
+					'count' => 2,
 				]
 			);
 			$list['activity']  = $activity_response['activity'];
+			$yesterday         = gmdate( 'Y-m-d', strtotime( '-1 day' ) );
+			
+			if ( ! empty( $activity_response['activity'][1] ) && $yesterday === $activity_response['activity'][1]['day'] ) {
+				$report->emails_sent  += $activity_response['activity'][1]['emails_sent'];
+				$report->opens        += $activity_response['activity'][1]['unique_opens'];
+				$report->clicks       += $activity_response['activity'][1]['recipient_clicks'];
+				$report->subscribes   += $activity_response['activity'][1]['subs'];
+				$report->unsubscribes += $activity_response['activity'][1]['unsubs'];
+			}
 		}
 
-		return self::process_data_to_report( $lists['lists'] );
+		return $report;
```

Comment:
Shouldn't this method should return the formatted report?

```suggestion
		return $report->to_array();
```

Reply:
I think it could return the object. Then we modify the consumer to handle the object instead of the array. But again, either way work fine

Reply:
Actually, my preference is to return the object. The whole point is to normalize the output of the report in all ESPs, and the signature enforces ESPs to return the object. If we let this method return an array, an ESP is not enforced to use that class.